_posts/2019-09-04-pulp-3-performance-blog.md

```diff
@@ -0,0 +1,88 @@
+
+## Test environment
+
+Pulp was installed on one CentOS7 (CentOS Linux release 7.5.1804 (Core)) system using this [playbook](https://github.com/pulp/ansible-pulp/blob/master/example-use/playbook.yml). 
+
+##### pulp version
+
+![version](https://user-images.githubusercontent.com/32102000/64325263-1a1c9380-cfc8-11e9-809c-210229c85887.png)
+
+In order to monitor the pulp3 performance, grafana monitoring has been done with the help of custom [script](https://github.com/redhat-performance/satellite-monitoring/blob/master/adhoc-scripts/get_stats_from_grafana.py). 
+
+##### grafana dashboard
+
+![grafana](https://user-images.githubusercontent.com/32102000/64325286-20ab0b00-cfc8-11e9-8ae4-0c92a475636b.png)
+
+## Tests
+
+| Tests | Run of |
+| ------------- | ------------- |
+| Single sync | 3 |
+| Concurrent sync  | 2  |
+| Single resync  | 3  |
+| Concurrent resync | 2  |
+| Single publish | 3  |
+| Concurrent publish  | 2  |
+| Single download  | 3  |
+| Concurrent download | 2  |
+| Listing repo and units | 4  |
+| Copy version | 2  |
+
+## Methodology
+
+Repos to test having files-count 100000 and created using this [script](https://github.com/redhat-performance/pulpperf/tree/master/scrips). In order to test the performance of pulp3, simple .py scripts are used and can be found [here](https://github.com/redhat-performance/pulpperf/tree/master/tests). Testing can be done on the pulp3 system using commands by specifying the test along with repos manually.
+
+For Example:  Single Sync pulp_file repo w/ 100k units
+
+Test Command:  ./sync_repository.py <path-of-the-repo>
+  
+#### Output 
+
+~~~
+Sync tasks
+==========
+/pulp/api/v3/tasks/4e04d054-e2ca-4a0e-8f88-3809a15d6484/
+    name:    pulp_file.app.tasks.synchronizing.synchronize
+    created:    2019-08-07T09:59:16.766175Z
+    started:    2019-08-07T09:59:16.897303Z
+    finished:    2019-08-07T11:05:00.733855Z
+    state:    completed
+
+   field                            min                            max
+   _created    2019-08-07T09:59:16.766175Z    2019-08-07T09:59:16.766175Z
+ started_at    2019-08-07T09:59:16.897303Z    2019-08-07T09:59:16.897303Z
+finished_at    2019-08-07T11:05:00.733855Z    2019-08-07T11:05:00.733855Z
+
+Sync tasks waiting time: {'samples': 1, 'min': 0.13, 'max': 0.13, 'mean': 0.13, 'stdev': 0.0}
+
+Sync tasks service time: {'samples': 1, 'min': 3943.84, 'max': 3943.84, 'mean': 3943.84, 'stdev': 0.0}
+~~~
+
+To get basic monitoring overview, we have used [get_stats_from_grafana.py](https://github.com/redhat-performance/satellite-monitoring/blob/master/adhoc-scripts/get_stats_from_grafana.py) script which connects to Grafana and formats some statistical measures. To get the more details on the output of all the commands, refer to this [document](https://docs.google.com/document/d/1Tt3WiQUaugYFkvuxVHkCjRngcMxqeYdMEKer_eqWOxE/edit#).
+
+#### Result-
+
+![results](https://user-images.githubusercontent.com/32102000/64327344-cf9d1600-cfcb-11e9-81eb-bec84adc11c8.png)
```

Comment:
Can we adjust this results section in two ways?

1) Can we drop the reporting on the system metrics, e.g. ram, cpu, etc. I think the main thing user want to understand is how long each operation takes. Can we report on the time and not on the system metrics?

2) Can we bring the results into this page instead of referring users to google docs? I'm concerned that anything that isn't on this page is not going to be read by the casual reader and with all of the results not here the blog post feels incomplete.